mvp/src/main/java/com/distillery/android/blueprints/mvp/architecture/BasePresenterProvider.kt

```diff
@@ -0,0 +1,9 @@
+package com.distillery.android.blueprints.mvp.architecture
+
+/**
+ * Presenter provider interface to create some presenters based on BasePresenter.
+ * Needs because koin can't work with generic types.
+ */
+interface BasePresenterProvider {
+    fun getPresenter(fragment: Presentable): BasePresenter
```

Comment:
No need to make the method name too verbose. It's clear that `BasePresenterProvider` provides a presenter so the name can be shortened:
```suggestion
    fun get(fragment: Presentable): BasePresenter
```

Reply:
I removed BasePresenterProvider because I found the way to inject it.